### PR TITLE
Feature/sinf 356 param tidy

### PR DESCRIPTION
**Pull request by adrianmilne**
This PR does a number of things:

1) Removes the use of the S3 bucket and client.env/spree.env files. The bucket and files are left in place for now for reference - to be deleted under a follow up ticket (SINF-371)

2) Removes any system parameters that can be inferred or set to sensible, overridable defaults (there are now just simple Terraform variables). Again - deletion of the orphaned system parameters to be done under SINF-371.

3) Converts any sensitive variables into ECS 'secrets' - which means they are not visible in the ECS console

4) Updated README

### CODE REVIEW

File: terraform/modules/configs/deploy-all/variables.tf [hunk 1]
```diff
@@ -6,48 +6,102 @@ variable "environment" {
...
+# (used to be param: /bat/{env}-logit-application)
+variable "logit_application" {
   type    = string
-  default = "latest"
+  default = null # this is purposfully null as we do a null check in the main.tf
 }
 
-variable "rollbar_env" {
+# (used to be env var: PAPERTRAIL_HOSTNAME from client.env)
+variable "papertrail_hostname" {
```
Comment (tmckernan) [inline terraform/modules/configs/deploy-all/variables.tf]: this papertrail envs can be removed,  as we now use logit.io for logging

File: terraform/modules/configs/deploy-all/variables.tf [hunk 2]
```diff
@@ -6,48 +6,102 @@ variable "environment" {
...
   type = string
+  default = "logs2.papertrailapp.com"
 }
 
-variable "client_cpu" {
-  type    = number
-  default = 2048
+# (used to be env var: PAPERTRAIL_REMOTE_PORT from client.env)
+variable "papertrail_remote_port" {
```
Comment (tmckernan) [inline terraform/modules/configs/deploy-all/variables.tf]: this papertrail envs can be removed,  as we now use logit.io for logging

File: terraform/modules/services/client/client.json.tpl [hunk 3]
```diff
@@ -95,6 +91,14 @@
       {
         "name": "LOGIT_APPLICATION",
         "value": "${logit_application}"
+      },
+      {
```
Comment (tmckernan) [inline terraform/modules/services/client/client.json.tpl:95]: this papertrail envs can be removed,  as we now use logit.io for logging

File: terraform/modules/services/client/ecs.tf [hunk 4]
```diff
@@ -90,21 +90,24 @@ data "template_file" "app_client" {
...
-    client_session_secret              = var.client_session_secret
-    logit_hostname                     = var.logit_hostname
-    logit_remote_port                  = var.logit_remote_port
     documents_terms_and_conditions_url = var.documents_terms_and_conditions_url
-    logit_node                         = var.logit_node
-    browser_rollbar_access_token       = var.browser_rollbar_access_token
     enable_basket                      = var.enable_basket
     enable_quotes                      = var.enable_quotes
     logit_application                  = var.logit_application
+    papertrail_hostname                = var.papertrail_hostname
```
Comment (tmckernan) [inline terraform/modules/services/client/ecs.tf]: this papertrail envs can be removed,  as we now using logit.io for logging

File: terraform/modules/services/client/variables.tf [hunk 5]
```diff
@@ -47,94 +43,100 @@ variable "spree_api_host" {
...
   type = string
 }
 
-variable "logit_hostname" {
+variable "logit_application" {
   type = string
 }
 
-variable "logit_remote_port" {
+variable "papertrail_hostname" {
```
Comment (tmckernan) [inline terraform/modules/services/client/variables.tf]: this papertrail envs can be removed,  as we now use logit.io for logging

File: terraform/modules/configs/deploy-all/variables.tf (hunk 1 above)
Comment (adrianmilne) [reply terraform/modules/configs/deploy-all/variables.tf]: Great - the less the merrier! I'll remove all those thanks

**Review by tmckernan — APPROVED** [review]
LGTM

File: README.md [hunk 6]
```diff
@@ -13,76 +13,40 @@ Check the README file for details of how to create the database.
...
+| /bat/{env}-basic-auth-username          | basic auth username for spree app and the client |
+| /bat/{env}-basic-auth-password          | basic auth password for spree app and the client |
+| /bar/{env}-basic-auth-enabled           | flag to enable/disable basic auth |
+| /bat/{env}-session-cookie-secret        | client session cookie secret |
+| /bat/{env}-products-import-bucket       | s3 bucket in which supplier import are progress |
+| /bat/{env}-logit-hostname               | logit.io endpoint |
+| /bat/{env}-logit-remote-port            | port to send logs to logit.io |
+| /bat/{env}-suppliers-sftp-bucket        | S3 bucket which holds suppliers sftp buckets |
+| /bat/{env}-sendgrid-api-key             | api key to sendgrid |
+| /bat/{env}-aws_access_key_id            | spree-user access key id |
```
Comment (tommyb82) [inline README.md]: Typo? `/bat/{env}-aws-access-key-id`

File: README.md [hunk 7]
```diff
@@ -13,76 +13,40 @@ Check the README file for details of how to create the database.
...
+| /bat/{env}-aws_secret_access_key        | spree-user secret access key |
```
Comment (tommyb82) [inline README.md]: Typo? `/bat/{env}-aws-secret-access-key`

File: README.md (hunk 6 above)
Comment (adrianmilne) [reply README.md]: Yep - hangover from old version - good spot!

File: README.md [hunk 8]
```diff
@@ -13,76 +13,40 @@ Check the README file for details of how to create the database.
...
-SESSION_COOKIE_SECRET={RANDOM_STRING? NOT SURE}
-DOCUMENTS_TERMS_AND_CONDITIONS_URL=https://purchasingplatform.crowncommercial.gov.uk/
-```
+| System Parameter                        | Description |
+| ----------------------------------------|-------------|
+| /bat/{env}-rollbar-access-token         | token to send errors to rollbar |
+| /bat/{env}-secret-key-base              | secret key for the spree app |
+| /bat/{env}-basic-auth-username          | basic auth username for spree app and the client |
+| /bat/{env}-basic-auth-password          | basic auth password for spree app and the client |
+| /bar/{env}-basic-auth-enabled           | flag to enable/disable basic auth |
```
Comment (tommyb82) [inline README.md:28]: Typo `/bar/` (unless it was that..) 😄

**Review by tommyb82 — APPROVED** [review]
Awesome stuff, applied to SBX2, all working great.  Good effort, makes things far, far simpler 😎 🎉

**Pull request merged**